lib/tasks/upgrade.rake

```diff
@@ -0,0 +1,39 @@
+# TRANSLATORS: do not translate
+desc <<-END_DESC
+  This task runs predefined upgrade steps.
+
+  Examples:
+    rake upgrade:run
+    rake upgrade:mark_as_ran TASK_NAME=some_rake_task,another_rake_task
+
+END_DESC
+
+namespace :upgrade do
+  task :run => :environment do
+    tasks = Foreman::Plugin.registered_plugins.values.map{|plugin| plugin.upgrade_tasks}.flatten
+    tasks.select!{|task| task.run?}
```

Comment:
Should they be sorted too or how is the order determined?

Reply:
right now they are sorted by:
1.  plugin order
2.  order they are defined in each plugin

I think 2) generally should cover everything within a plugin, but i could see 1) causing a problem since we don't have much control over that.  If you needed to run some task before a task in another plugin there is no mechanism to do that.  I don't expect that to be a problem, and i think we could tackle that complication if we ever did come to it.